qa/rpc-tests/zkey_import_export.py

```diff
@@ -0,0 +1,185 @@
+            results = node.z_getoperationresult(opids)
+            if len(results)==0:
+                time.sleep(1)
+            else:
+                print("Results", results[0])
+                status = results[0]["status"]
+                if status == "failed":
+                    errormsg = results[0]['error']['message']
+                elif status == "success":
+                    txid = results[0]['result']['txid']
+                break
+        print('...returned status: {}'.format(status))
+        assert_equal(in_status, status)
+        if errormsg is not None:
+            assert(in_errormsg is not None)
+            assert_equal(in_errormsg in errormsg, True)
+            print('...returned error: {}'.format(errormsg))
+        return txid
+
+    def run_test(self):
+        [alice, bob, charlie, miner] = self.nodes
+
+        def alice_to_bob(amount):
+            print("Alice zaddr in alice to bob", alice_zaddr)
+            opid = alice.z_sendmany(alice_zaddr, [{"address": bob_zaddr, "amount": Decimal(amount)}])
+            txid = self.wait_and_assert_operationid_status(alice, opid)
+            print("Txid of z_sendmany", txid)
+            self.sync_all()
+            miner.generate(1)
+            self.sync_all()
+
+        def verify_utxos(node, amts, zaddr):
+            amts.sort(reverse=True)
+            txs = node.z_listreceivedbyaddress(zaddr)
+
+            def cmp_confirmations_high_to_low(a, b):
+                return cmp(b["amount"], a["amount"])
+
+            txs.sort(cmp_confirmations_high_to_low)
+            print("Sorted txs", txs)
+            print("amts", amts)
+
+            try:
+                assert_equal(amts, [tx["amount"] for tx in txs])
+            except AssertionError:
+                logging.error(
+                    'Expected amounts: %r; txs: %r',
+                    amts, txs)
+                raise
+
+        def get_private_balance(node):
+            balance = node.z_gettotalbalance()
+            return balance['private']
+
+        def find_imported_key(node, import_zaddr):
+            zaddrs = node.z_listaddresses()
+            for zdr in zaddrs:
+                if zdr == import_zaddr:
+                    return zdr
+                else:
+                    print("Imported zaddr {0} not found".format(import_zaddr))
+
+        # Seed Alice with some funds
+        alice.generate(10)
+        self.sync_all()
+        miner.generate(100)
+        self.sync_all()
+        # Shield Alice's coinbase funds to her zaddr
+        alice_zaddr = alice.z_getnewaddress()
+        res = alice.z_shieldcoinbase("*", alice_zaddr)
+        txid = self.wait_and_assert_operationid_status(alice, res['opid'])
+        miner.generate(6)
+        self.sync_all()
+        # List funds
+        funds = alice.z_listreceivedbyaddress(alice_zaddr)
+        # print("Alice's funds after shield", funds)
+
+        # Ensure new funds have been confirmed
+        miner.generate(2)
```

Comment:
What does this step accomplish that the `miner.generate()` call before `alice.z_listreceivedbyaddress()` does not?

Reply:
... you're right, I was using this earlier but now it's unnecessary.